src/dependency-manager/src/utils/interpolation/parameter-interpolator.ts

```diff
@@ -0,0 +1,119 @@
+import Mustache from 'mustache';
+import { ServiceNode } from '../..';
+import DependencyGraph from '../../graph';
+import { InterpolationContext } from '../../interpolation/interpolation-context';
+import { ParameterValueV2 } from '../../service-config/base';
+import { EnvironmentInterfaceContext, EnvironmentInterpolationContext, ServiceInterfaceContext } from './interpolation-context';
+
+export class ParameterInterpolator {
+
+  public static interpolateString(param_value: string, environment_context: EnvironmentInterpolationContext): string {
+    Mustache.tags = ['${', '}']; // sets custom delimiters
+    Mustache.escape = function (text) { return text; } // turns off HTML escaping
+    //TODO:77: add validation logic to catch expressions that don't refer to an existing path
+    return Mustache.render(param_value, environment_context);
+  }
+
+  public static build_friendly_name_map(graph: DependencyGraph): { [key: string]: { [key: string]: string } } {
+    const friendly_name_map: { [key: string]: { [key: string]: string } } = {};
+    for (const node of graph.nodes) {
+      if (node instanceof ServiceNode) {
```

Comment:
Need to account for ExternalNodes. Add a test case to cover.

Reply:
Like I said above I'm okay not supporting external nodes as a first pass and hopefully they would be supported out of the box by:
https://gitlab.com/architect-io/architect-cli/-/issues/89